### PR TITLE
Wait for config to be ready before accessing values

### PR DESCRIPTION
**Pull request by mattseddon**
I saw lots of type errors being sent to the integration test logs. After investigating I found and fixed some issues.

### CODE REVIEW

File: extension/src/test/suite/setup/index.test.ts [hunk 1]
```diff
@@ -577,7 +577,7 @@ suite('Setup Test Suite', () => {
       mockExecuteCommand.restore()
       mockRunSetup.restore()
       stub(config, 'isPythonExtensionUsed').returns(false)
-      stub(config, 'getPythonBinPath').resolves(join('python'))
+      stub(config, 'getPythonBinPath').returns(join('python'))
```
Comment (mattseddon) [inline extension/src/test/suite/setup/index.test.ts:580]: [F] This was the cause of [these test warnings](https://github.com/iterative/vscode-dvc/actions/runs/4856306390/jobs/8655732780#step:7:677).

File: extension/src/setup/index.ts [hunk 2]
```diff
@@ -389,18 +390,14 @@ export class Setup
     }
   }
 
-  private isDVCBeingUsedGlobally() {
```
Comment (mattseddon) [inline extension/src/setup/index.ts:392]: We append the path to python to the `PATH` variable for the edge case that DVC is being used as a binary but with a virtual environment.

File: extension/src/setup/index.ts [hunk 3]
```diff
@@ -374,7 +374,8 @@ export class Setup
     return this.sendDataToWebview()
   }
 
-  public getDvcCliDetails(): DvcCliDetails {
+  private async getDvcCliDetails(): Promise<DvcCliDetails> {
+    await this.config.isReady()
```
Comment (mattseddon) [inline extension/src/setup/index.ts:378]: [F] Need to make sure that the config has set the pythonBinPath before we try to access it.

**Review by sroy3 — APPROVED** [review]

File: extension/src/setup/index.ts (hunk 2 above)
Comment (mattseddon) [reply extension/src/setup/index.ts:392]: because we append the virtual environment path to the PATH environment variable it is technically being used.

**Pull request merged**